server/api.go

```diff
@@ -276,6 +276,31 @@ func (s *ApiServer) Healthcheck(ctx context.Context, in *empty.Empty) (*empty.Em
 
 func securityInterceptorFunc(logger *zap.Logger, config Config, ctx context.Context, req interface{}, info *grpc.UnaryServerInfo) (context.Context, error) {
 	switch info.FullMethod {
+	case "/nakama.api.Nakama/AuthenticateRefresh":
+		md, ok := metadata.FromIncomingContext(ctx)
+		if !ok {
+			logger.Error("Cannot extract metadata from incoming context")
+			return nil, status.Error(codes.FailedPrecondition, "Cannot extract metadata from incoming context")
+		}
+		auth, ok := md["authorization"]
+		if !ok {
+			return nil, status.Error(codes.FailedPrecondition, "Auth token required")
+		}
+
+		if len(auth) != 1 {
+			// Value of "authorization" or "grpc-authorization" was empty or repeated.
+			return nil, status.Error(codes.Unauthenticated, "Auth token invalid")
+		}
+
+		//TODO check re-use and FailPrecondition
+
+		userID, username, vars, exp, ok := parseBearerAuth([]byte(config.GetSession().RefreshEncryptionKey), auth[0])
```

Comment:
I think instead of passing this token as part of the `Authorization` header we should make it part of the request body, and require the server key in the `Authorization` header like the other authentication endpoints.